src/AzureRenderManager/WebApp/Controllers/ReportingController.cs

```diff
@@ -123,29 +131,39 @@ public static QueryTimePeriod ThisMonth()
             return new QueryTimePeriod(from: StartOfMonth(today), to: EndOfMonth(today));
         }
 
-        private static async Task<(string envName, UsageResponse)> GetUsage(
+        private async Task<(string envName, UsageResponse)> GetUsage(
             CostManagementClient client,
             RenderingEnvironment env,
             QueryTimePeriod timePeriod)
         {
-            var usageRequest =
-                new UsageRequest(
-                    Timeframe.Custom,
-                    new Dataset(
-                        Granularity.Daily,
-                        new Dictionary<string, Aggregation>
-                        {
+            var cacheKey = Invariant($"Reporting/{env.Name}/{timePeriod.From}/{timePeriod.To}");
+
+            var usageResponse = await _memoryCache.GetOrCreateAsync(cacheKey, Fetch); 
+            return (env.Name, usageResponse);
+
+            Task<UsageResponse> Fetch(ICacheEntry ice)
+            {
+                ice.AbsoluteExpirationRelativeToNow = CacheResultsFor;
+                
+                var usageRequest =
+                    new UsageRequest(
+                        Timeframe.Custom,
+                        new Dataset(
+                            Granularity.Daily,
+                            new Dictionary<string, Aggregation>
+                            {
                             { "totalCost", new Aggregation(AggregationFunction.Sum, "PreTaxCost") }
-                        },
-                        new List<Grouping>
-                        {
+                            },
+                            new List<Grouping>
+                            {
                             new Grouping("MeterSubCategory", ColumnType.Dimension)
-                        },
-                        FilterExpression.Tag("environment", Operator.In, new[] { env.Id })));
+                            },
+                            FilterExpression.Tag("environment", Operator.In, new[] { env.Id })));
 
-            usageRequest.TimePeriod = timePeriod;
+                usageRequest.TimePeriod = timePeriod;
 
-            return (env.Name, await client.GetUsageForResourceGroup(env.SubscriptionId, env.ResourceGroupName, usageRequest));
+                return client.GetUsageForResourceGroup(env.SubscriptionId, env.ResourceGroupName, usageRequest);
```

Comment:
Missed this earlier - resources could be spread across many resource groups.  For example the storage server is in a dedicated RG.  We'll need to get the subscription costs, filtered by tag.

Reply:
I'll address this in another PR.